docs/zarr-v3.md

```diff
@@ -0,0 +1,96 @@
+            dataset_kwargs["chunks"] = tuple(i // 2 for i in dataset_kwargs["shards"])
+        elif iospec.encoding_type in { "csr_matrix", "csc_matrix" }:
+            dataset_kwargs = { "shards": (2**16,), "chunks": (2**8, ), **dataset_kwargs }
+        func(g, k, elem, dataset_kwargs=dataset_kwargs)
+    return ad.experimental.write_dispatched(group, "/", adata, callback=callback)
+```
+
+However, `zarr-python` can be slow with sharding throughput as well as writing throughput.
+Thus if you wish to speed up either writing, sharding, or both (or receive a modest speed-boost for reading), a bridge to the `zarr` implementation in Rust: https://zarrs-python.readthedocs.io/en/latest/ can help with that (see https://github.com/LDeakin/zarr_benchmarks for benchmarks):
+
+```
+uv pip install zarrs
+```
+
+```python
+import zarr
+import zarrs
+zarr.config.set({"codec_pipeline.path": "zarrs.ZarrsCodecPipeline"})
+```
+
+However, this pipeline is not compatible with all types of zarr store, especially remote stores and there are limitations on where rust can give a performance boost for indexing.
+We therefore recommend this pipeline for writing full datasets and reading contiguous regions of said written data.
+
+## Codecs
+
+The default `zarr-python` v3 codec for the [v3 format] is no longer `blosc` but `zstd`.
+While `zstd` is more widespread, you may find its performance to not meet your old expectations.
+Therefore, we recommend passing in the [`BloscCodec`] to `compressor` on {func}`~anndata.AnnData.write_zarr` if you wish to return to the old behavior.
+
+There is currently a bug with `numcodecs` that prevents data written from other non-numcodecs `zstd` implementations from being read in by the default zarr pipeline (to which the above rust pipeline falls back if it cannot handle a datatype or indexing scheme, like `vlen-string`): https://github.com/zarr-developers/numcodecs/issues/424.
+Thus is may be advisable to use `BloscCodec` with `zarr` v3 file format data if you wish to use the rust-accelerated pipeline until this issue is resolved.
+
+The same issue with `zstd` applies to data that may eventually be written by the GPU `zstd` implementation (see below).
+
+## GPU i/o
+
+At the moment, it is unlikely your `anndata` i/o will work if you use [`zarr.enable_gpu`].
+It's *possible* dense data i/o i.e., using {func}`anndata.io.read_elem` will work as expected, but this functionality is untested - sparse data, awkward arrays, and dataframes will not.
+`kvikio` currently provides a [`GDS`-enabled store] although there are no working compressors at the moment exported from the `zarr-python` package (work is underway for `Zstd`: https://github.com/zarr-developers/zarr-python/pull/2863).
+
+We anticipate enabling officially supporting this functionality officially for dense data, sparse data, and possibly awkward arrays in the next minor release, 0.13.
+
+## Asynchronous i/o
+
+At the moment, `anndata` exports no `async` functions.
```

Comment:
That’ll change soon with #1902, right? So maybe we should merge this first, then make sure this line is changed in that PR.